src/celengine/frametree.cpp

```diff
@@ -172,23 +170,22 @@ FrameTree::addChild(TimelinePhase* phase)
  *  phase doesn't exist in the tree.
  */
 void
-FrameTree::removeChild(TimelinePhase* phase)
+FrameTree::removeChild(shared_ptr<const TimelinePhase> &phase)
 {
-    vector<TimelinePhase*>::iterator iter = find(children.begin(), children.end(), phase);
+    auto iter = find(children.begin(), children.end(), phase);
     if (iter != children.end())
     {
-        (*iter)->release();
         children.erase(iter);
         markChanged();
     }
 }
 
 
 /*! Return the child at the specified index. */
-TimelinePhase*
+const TimelinePhase*
```

Comment:
why do you have raw pointer here?

i suppose we *should* always return smart pointers, we *can* accept raw pointers (if we don't store them anywhere)

Reply:
Probably because I wanted to check impact on perfomance on the first place. Now I'm implementing it more comprehently and I'm stuck with domino effect... Probably will be finished tomorrow.